src/lib.rs

```diff
@@ -44,12 +44,26 @@ where
 
 impl<T> Merge for Vec<T>
 where
-    T: Hash + Eq,
+    T: Hash + Eq + Clone,
 {
-    fn merge(self, other: Self) -> Self {
-        let mut merged = self;
-        merged.extend(other);
-        merged.deduplicate()
+    fn merge(mut self, other: Self) -> Self {
+        use std::collections::HashMap;
+
+        let mut position_map: HashMap<T, usize> = self
+            .iter()
+            .enumerate()
+            .map(|(index, item)| (item.clone(), index))
+            .collect();
+
+        for item in other {
```

Comment:
Iterate over other, which is small array, complexity O(m).

Reply:
This is being used in full updates as well, so will be dangerous, why are we changing the merge method?